Code/Framework/AzCore/AzCore/Serialization/AZStdContainers.inl

```diff
@@ -48,6 +48,16 @@ namespace AZStd
     class intrusive_ptr;
     template<class T>
     class shared_ptr;
+
+    namespace AssociativeInternal
+    {
+        // SFINAE expression to determine whether an associative container is an actual map with a corresponding value for each key,
+        // or is simply a keyed container, like a set or an unordered_set
+        template<class T, class = void>
+        constexpr bool IsMapType_v = false;
+        template<class T>
+        constexpr bool IsMapType_v<T, enable_if_t<Internal::sfinae_trigger_v<typename T::mapped_type>>> = true;
```

Comment:
Nice!

Reply:
yep! Courtesy of our guru @lumberyard-employee-dm  👍